lib/tasks/dial.js

```diff
@@ -468,6 +471,52 @@ class TaskDial extends Task {
     this.dials.clear();
   }
 
+  /**
+   * Check if two SDP endpoints are using the same primary audio codec
+   * @param {object} ep1 - First endpoint object
+   * @param {object} ep2 - Second endpoint object
+   * Sets this.mediaNotSame to false if same codec (no transcoding needed), true otherwise
+   */
+  _checkTranscoding(ep1, ep2) {
```

Comment:
lets create a new file utils/sdp-utils.js and place a function there `getLeadingCodec(sdp)` which parses an SDP and provides the first audio codec in the list